libff/algebra/curves/tests/test_groups.cpp

```diff
@@ -317,6 +317,26 @@ void test_bls12_377()
     ASSERT_EQ(bls12_377_G2::zero(), z);
 }
 
+// check that some elements e.g. 1,-1,2,random satisfy the curve
+// equation Y^2 = X^3 + a X + b; used in test_bls12_381
+template<typename GroupT> void check_curve_equation(GroupT P)
+{
+    P.to_affine_coordinates();
+    using Fq = typename std::decay<decltype(P.X)>::type;
+    Fq lhs = (P.Y * P.Y);
+    Fq rhs = ((P.X * P.X * P.X) + (GroupT::coeff_a * P.X) + GroupT::coeff_b);
+    ASSERT_EQ(lhs, rhs);
+}
+
+template<typename GroupT> void test_bls12_381()
```

Comment:
Let's rename this to something like `test_curve_equation` and run for each curve.

(https://github.com/clearmatics/libff/pull/68/files#discussion_r859996573)

Reply:
I have added the `test_curve_equation()` check to all curves except Edwards and BN128. For the latter two I get a compilation error since they don't have a member `coeff_a` .